Gemfile

```diff
@@ -13,7 +13,14 @@ gem 'bundler'
 gem 'rake'
 
 group :system_tests do
-  gem 'beaker'
+  beaker_version = ENV.fetch('BEAKER_VERSION', ['>= 4.17.0', '< 5.0.0'])
+
+  if "#{beaker_version}".include?('/')
+    gem 'beaker', :source => beaker_version
```

Comment:
* Using `:source` enables folks to specify an alternative rubygems repo, but it wouldn't permit them to specify an arbitrary git repo or ref (which in this instance seems like the more likely use case).

* Also, doubling up `BEAKER_VERSION` to be either a URL or version number has the side effect that you can only specify the location or the version―but not both.

Reply:
@op-ct Hrm...can you think of a good way of getting all permutations in there without having 5 million variables?